### PR TITLE
Client credentials scope

### LINKED ISSUE #4: Optional scope in client_credentials
Some auth servers (e.g., Azure Entra ID) do not fall back to a default scope and instead fail the token request when a scope is omitted. This makes users rely on the `token_request_builder` to build up the request instead of using the `client_credentials` function. It would be great to have an optional `scope` parameter or another dedicated function.

#### Possible options
In both cases, the scope will be added to the request builder's params and request according to [RFC6749 &sect;3.3][0] and [&sect;4.4.2][1].

##### 

##### Optional `scope` parameter (breaking change)

```
pub fn client_credentials(
  client: Client(body),
  token_uri: UriAppendage,
  auth_scheme: AuthScheme,
  scope: Option(Scope)
) -> Request(String) { ... }
```

##### Dedicated function with scope

```
pub fn scoped_client_credentials(
  client: Client(body),
  token_uri: UriAppendage,
  auth_scheme: AuthScheme,
  scope: Scope
) -> Request(String) { ... }
```

##### Scope Record?
```
/// Represents the scope for a token request.
pub type Scope {
  /// Respresents a scope as a list of individual strings.
  ScopeList(List(String))
  /// Represents a scope as a string of one or more space-deliminated scopes.
  /// Useful for a single scope or when the scopes have been pre-joined.
  ScopeString(String)
}
```

Perhaps, you can think of other better options. I'm happy to implement the change based on your suggestions and feedback.

[0]: https://datatracker.ietf.org/doc/html/rfc6749#section-3.3
[1]: https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2

### PR DESCRIPTION
**Pull request by ChiefTwoPencils**
Adds the ability to provide a scope to a client credentials.
- Adds `ScopeList`, `ScopeString`, and `DefaultScope` types.
- Adds test for all scope types.
- Adds usage example for client credentials flow.
- Closes #4.

### CODE REVIEW

File: src/glow_auth/token_request.gleam [hunk 1]
```diff
@@ -33,18 +34,35 @@ pub fn authorization_code(
...
+/// The [Access Token Scope](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3)
+/// is a list of unordered, space-deliminated, case-sensitive strings.
+pub type Scope {
+  /// Respresents the access token scope as a list of individual scope values that can 
+  /// later be joined into a space-delimited string.
+  ScopeList(List(String))
+  /// Represents the access token scope as a string of one or more space-deliminated scopes.
+  /// Useful for a single scope or when the scopes have been pre-joined.
+  ScopeString(String)
+  /// Represents the omittion of an access token scope.
```
Comment (adz) [inline src/glow_auth/token_request.gleam]: typo? omittion => omission.

File: .tool-versions [hunk 2]
```diff
@@ -1,3 +1,3 @@
 erlang 26.2.5
-gleam 1.2.1
```
Comment (adz) [inline .tool-versions]: maybe don't update gleam in same PR - I'll do another one to 1.4 which just launched.

File: examples.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    Ok(wrapped) -> decode_token_from_response(wrapped.body)
+    _ -> panic
+  }
+
+  // View the Result token in the console.
+  io.debug(token)
+  |> Ok
+}
+```
+
```
Comment (adz) [inline examples.md:79]: Lovely :sparkles:

File: src/glow_auth/token_request.gleam (hunk 1 above)
Comment (ChiefTwoPencils) [reply src/glow_auth/token_request.gleam]: Indeed, nice catch.

File: .tool-versions (hunk 2 above)
Comment (ChiefTwoPencils) [reply .tool-versions]: Sounds good. Thought I'd throw it in there to see what you preferred.

**Comment by adz** [comment]
Thankyou!

**Pull request merged**